modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1837,18 +1821,20 @@ public CompletableFuture<List<Table>> tablesAsync() {
      * @return Future representing pending completion of the operation.
      */
     private CompletableFuture<List<Table>> tablesAsyncInternal() {
-        return supplyAsync(() -> inBusyLock(busyLock, this::directTableIds), ioExecutor)
-                .thenCompose(tableIds -> inBusyLock(busyLock, () -> {
-                    var tableFuts = new CompletableFuture[tableIds.size()];
+        return schemaSyncService.waitForMetadataCompleteness(clock.now())
+                .thenCompose(ignore -> inBusyLock(busyLock, () -> {
+                    NamedListView<TableView> configuredTables = tablesCfg.tables().value();
```

Comment:
Ahem, now the method goes go the configuration. Wouldn't the approach 'take tables from tablesByIdVv, then for each of them wait till they are ready and then return the tables' work? It seems that it wouldn't require to go to the configuration

Reply:
Fixed.
Seems, we shouldn't either use configuration here or mute tests.
E.g. `TableManagerTest.testGetTableDuringCreation`failed on TC because TableManager triggered event before Catalog being updated.